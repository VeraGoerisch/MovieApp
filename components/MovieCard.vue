<template>
  <div class="card">
    <img :src="imageSrc" :alt="`${title}-image`" />
    <p class="title">{{ title }}</p>
  </div>
</template>

<script>
export default {
  props: {
    imageSrc: {
      type: String,
      required: true,
    },
    title: {
      type: String,
      required: true,
    },
  },
};
</script>

<style lang="postcss">
.card {
  width: 200px;
  margin: 20px 10px;
  padding: 10px;
  &:hover {
    border-radius: 5px;
    box-shadow: -2px 2px 15px 2px rgba(0, 0, 0, 0.67);
  }
  & img {
    width: 185px;
    height: 278px;
  }
  & .title {
    font-weight: bold;
    margin: 5px 0 10px;
  }
}
</style>
