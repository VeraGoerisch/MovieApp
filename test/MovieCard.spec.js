import { mount } from '@vue/test-utils';
import MovieCard from '@/components/MovieCard.vue';

describe('MovieCard', () => {
  test('this is a vue instance', () => {
    const wrapper = mount(MovieCard, {
      propsData: {
        imageSrc: 'testImage.jpg',
        title: 'Test Title',
      },
    });
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = mount(MovieCard, {
      propsData: {
        imageSrc: 'testImage.jpg',
        title: 'Test Title',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
