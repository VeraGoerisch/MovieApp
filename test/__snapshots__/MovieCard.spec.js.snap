// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders properly 1`] = `
"<div class=\\"card\\"><img src=\\"testImage.jpg\\" alt=\\"Test Title-image\\">
  <p class=\\"title\\">Test Title</p>
</div>"
`;
